packages/@aws-cdk/aws-efs/lib/filesystem.ts

```diff
@@ -0,0 +1,130 @@
+     */
+    performanceMode?: PerformanceMode
+}
+/* tslint:enable:max-line-length */
+
+/**
+ * Provides a Elastic File System (EFS).
+ */
+export class FileSystem extends Construct {
+
+    public readonly fileSystemId: FileSystemId;
+
+    private readonly fileSystem: cloudformation.FileSystemResource;
+
+    constructor(parent: Construct, id: string, props: FileSystemProps = {}) {
+        super(parent, id);
+
+        const throughput = props.throughputMiB === undefined ? 100 : props.throughputMiB;
+        if (throughput < 1.0) {
+            throw new Error(`Provisioned throughput can\'t be set to a value less than 1.0 MiB/s, provided value: ${props.throughputMiB} `);
+        }
+        const throughputMode = props.throughputMode || ThroughputMode.Bursting;
+        const performanceMode = props.performanceMode || PerformanceMode.GeneralPurpose;
+
+        this.fileSystem = new cloudformation.FileSystemResource(this, 'Resource', {
+            encrypted: true,
+            performanceMode,
+            provisionedThroughputInMibps: throughput,
+            throughputMode
+        });
+
+        this.fileSystemId = this.fileSystem.ref;
+    }
+}
+
+/**
+ * The throughput mode for the file system to be created. There are two throughput modes to choose
+ * from for your file system: bursting and provisioned. You can decrease your file system's
+ * throughput in Provisioned Throughput mode or change between the throughput modes as long as it’s
+ * been more than 24 hours since the last decrease or throughput mode change.
+ */
+export enum ThroughputMode {
+    Bursting = 'bursting',
+    Provisioned = 'provisioned',
+}
+
+/**
+ * The PerformanceMode of the file system. We recommend generalPurpose performance mode for most
+ * file systems. File systems using the maxIO performance mode can scale to higher levels of
+ * aggregate throughput and operations per second with a tradeoff of slightly higher latencies for
+ * most file operations. This can't be changed after the file system has been created.
+ */
+export enum PerformanceMode {
+    GeneralPurpose = 'generalPurpose',
+    MaxIO = 'maxIO',
+}
+
+export abstract class FileSystemRef extends Construct {
+    public static import(parent: Construct, id: string, props: FileSystemRefProps) {
+      return new ImportedFileSystem(parent, id, props);
+    }
+
+    public abstract readonly fileSystemId: FileSystemId;
+    constructor(parent: Construct, id: string) {
+      super(parent, id);
+    }
+
+    public export(): FileSystemRefProps {
+      return {
+        fileSystemId: new FileSystemId(new Output(this, 'FileSystemId', { value: this.fileSystemId }).makeImportValue())
+      };
+    }
+}
+
+export class ImportedFileSystem extends FileSystemRef {
+    public readonly fileSystemId: FileSystemId;
+    constructor(parent: Construct, id: string, props: FileSystemRefProps) {
+      super(parent, id);
+      this.fileSystemId = props.fileSystemId;
+    }
+  }
```

Comment:
Indentation

Reply:
you fixed this when you updated to the new indentation.